nym-vpn-desktop/installer/appimage.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+# NymVPN AppImage installer
+
+set -E
+set -o pipefail
+trap 'catch $? ${FUNCNAME[0]:-main} $LINENO' ERR
+cwd=$(pwd)
+
+# ANSI style codes
+RED="\e[38;5;1m" # red
+GRN="\e[38;5;2m" # green
+YLW="\e[38;5;3m" # yellow
+BLD="\e[1m"      # bold
+RS="\e[0m"       # style reset
+B_RED="$BLD$RED"
+B_GRN="$BLD$GRN"
+B_YLW="$BLD$YLW"
+####
+
+# TODO replace with the current tag and version
+tag=test-appimage
+version=0.0.8-dev
+appimage_url="https://github.com/nymtech/nym-vpn-client/releases/download/$tag/nym-vpn_${version}.AppImage"
+# TODO disable the desktop entry install, until the app can run
+# without root privileges
+desktop_entry_disabled=true
+
+catch() {
+  log_e "$B_RED✗$RS unexpected error, [$BLD$1$RS] $BLD$2$RS L#$BLD$3$RS"
+  cleanup
+  cd "$cwd" || true
+  exit 1
+}
+
+log() {
+  echo -e "$1"
+}
+
+log_e() {
+  >&2 echo -e "$1"
+}
+
+pushd() {
+  command pushd "$@" > /dev/null || exit 1
+}
+
+popd() {
+  command popd > /dev/null || exit 1
+}
+
+need_cmd() {
+  if ! command -v "$1" > /dev/null 2>&1; then
+    log_e " $B_RED⚠$RS need$BLD $1$RS (command not found)"
+    exit 1
+  fi
+}
+
+need_cmd mktemp
+temp_dir=$(mktemp -d)
+install_dir="$HOME/.local/bin"
+icons_dir="$HOME/.local/share/icons"
+appimage="nym-vpn_${version}.AppImage"
+target_appimage="nym-vpn.appimage"
+desktop_dir="$HOME/.local/share/applications"
```

Comment:
What about looking these directories up by referring to XDG? Not sure we can assume that they will exist

Reply:
`~/.local/share` is very standard and `XDG_DATA_HOME` default. Even if the sub dirs don't exist, `install -D...` will create them as needed.

Reply:
but yes we can check if `$XDG_DATA_HOME` is set first and default to `$HOME/.local/share` if not (according to XDG base dir spec)

Reply:
Regarding `$HOME/.local/bin` there is no XDG variable for it, but it is specified in the spec

> User-specific executable files may be stored in $HOME/.local/bin. Distributions should ensure this directory shows up in the UNIX $PATH environment variable, at an appropriate place.

see https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html